torch/ao/sparsity/sparsifier.py

```diff
@@ -0,0 +1,182 @@
+        """
+        module_groups = [
+            dict(filter(lambda key_value: key_value[0] != 'module', mg.items()))
+            for mg in self.module_groups
+        ]
+        return {
+            'module_groups': module_groups,
+        }
+
+    def load_state_dict(self, state_dict, strict=True):
+        module_groups = copy.deepcopy(state_dict['module_groups'])
+        for group in module_groups:
+            layer = _fqn_to_module(self.model, group['path'])
+            if strict and layer is None:
+                raise RuntimeError(f'Error loading group["path"] into the model')
+            group['module'] = layer
+        self.__setstate__({'module_groups': module_groups})
+
+    def prepare(self, model, config):
+        r"""Prepares a model, by adding the parametrizations.
+
+        Note::
+
+            The model is modified inplace. If you need to preserve the original
+            model, use copy.deepcopy.
+        """
+        self.config = config
+        self.model = model
+        # If no config -- try getting all the supported layers
+        if self.config is None:
+            # Add all models to the config
+            self.config = []
+            stack = [model]
+            while stack:
+                module = stack.pop()
+                for name, child in module.named_children():
+                    if type(child) in SUPPORTED_MODULES:
+                        self.config.append(child)
+                    else:
+                        stack.append(child)
+
+        for module_config in self.config:
+            if isinstance(module_config, nn.Module):
+                module_config = {'module': module_config}
+            local_args = copy.deepcopy(self.defaults)
+            local_args.update(module_config)
+            module = local_args['module']
+            module_path = _module_to_fqn(self.model, module)
+            if module_path and module_path[0] == '.':
+                module_path = module_path[1:]
+            local_args['path'] = module_path
+            self.module_groups.append(local_args)
+
+        self._prepare()
+
+    def _prepare(self, *args, **kwargs):
+        r"""Adds mask parametrization to the layer weight
+        """
+        for config in self.module_groups:
+            module = config['module']
+            if getattr(module, 'mask', None) is None:
+                module.register_buffer('mask', torch.ones(module.weight.shape))
+            param = config.get('parametrization', FakeSparsity)
+            parametrize.register_parametrization(module, 'weight',
+                                                 param(module.mask))
+
+    def squash_mask(self, *args, **kwargs):
+        for config in self.module_groups:
+            module = config['module']
+            parametrize.remove_parametrizations(module, 'weight',
+                                                leave_parametrized=True)
+            if getattr(module._parameters, 'mask', None):
```

Comment:
Because this is a generalized approach, we must consider a case where the mask is part of the module. I will add a check if there is an attribute in the module